Compile Error with Hazelcast 3.12-SNAPSHOT

There is a breaking change coming with that [PR](https://github.com/hazelcast/hazelcast/pull/14262)
A new hazelcast-aws needs to be release together with 3.12-BETA and onwards...



http://jenkins.hazelcast.com/view/Plugins/job/AWS-master/14/consoleFull


```
19:01:47 /home/jenkins/jenkins_slave/workspace/AWS-master/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java:[143,32] error: cannot find symbol
19:01:47 [[1;34mINFO[m] 6 errors 
19:01:47 [[1;34mINFO[m] -------------------------------------------------------------
19:01:47 [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
19:01:47 [[1;34mINFO[m] [1;31mBUILD FAILURE[m
19:01:47 [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
19:01:47 [[1;34mINFO[m] Total time: 35.691 s
19:01:47 [[1;34mINFO[m] Finished at: 2019-02-04T00:01:47Z
19:01:47 [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
19:01:47 [[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile[m [1m(default-compile)[m on project [36mhazelcast-aws[m: [1;31mCompilation failure[m: Compilation failure: 
19:01:47 [[1;31mERROR[m] /home/jenkins/jenkins_slave/workspace/AWS-master/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java:[34,0] error: cannot find symbol
19:01:47 [[1;31mERROR[m]   symbol:   static cleanNodeName
19:01:47 [[1;31mERROR[m]   location: class
19:01:47 [[1;31mERROR[m] /home/jenkins/jenkins_slave/workspace/AWS-master/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java:[33,0] error: cannot find symbol
19:01:47 [[1;31mERROR[m]   symbol:   static childElements
19:01:47 [[1;31mERROR[m]   location: class
19:01:47 [[1;31mERROR[m] /home/jenkins/jenkins_slave/workspace/AWS-master/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java:[129,30] error: cannot find symbol
19:01:47 [[1;31mERROR[m]   symbol:   method childElements(Node)
19:01:47 [[1;31mERROR[m]   location: class NodeHolder
19:01:47 [[1;31mERROR[m] /home/jenkins/jenkins_slave/workspace/AWS-master/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java:[130,32] error: cannot find symbol
19:01:47 [[1;31mERROR[m]   symbol:   method cleanNodeName(Node)
19:01:47 [[1;31mERROR[m]   location: class NodeHolder
19:01:47 [[1;31mERROR[m] /home/jenkins/jenkins_slave/workspace/AWS-master/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java:[142,30] error: cannot find symbol
19:01:47 [[1;31mERROR[m]   symbol:   method childElements(Node)
19:01:47 [[1;31mERROR[m]   location: class NodeHolder
19:01:47 [[1;31mERROR[m] /home/jenkins/jenkins_slave/workspace/AWS-master/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java:[143,32] error: cannot find symbol
19:01:47 [[1;31mERROR[m] -> [1m[Help 1][m
```

--- a/pom.xml
+++ b/pom.xml
@@ -29,7 +29,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
         <jdk.version>1.6</jdk.version>
-        <hazelcast.version>3.11.1</hazelcast.version>
+        <hazelcast.version>3.12-BETA-1</hazelcast.version>
 
         <junit.version>4.12</junit.version>
         <mockito.version>1.10.19</mockito.version>
@@ -220,6 +220,7 @@
             <groupId>com.hazelcast</groupId>
             <artifactId>hazelcast</artifactId>
             <version>${hazelcast.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.hazelcast</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -29,7 +29,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
         <jdk.version>1.6</jdk.version>
-        <hazelcast.version>3.11</hazelcast.version>
+        <hazelcast.version>3.11.1</hazelcast.version>
 
         <junit.version>4.12</junit.version>
         <mockito.version>1.10.19</mockito.version>

--- a/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java
+++ b/src/main/java/com/hazelcast/aws/utility/CloudyUtility.java
@@ -30,8 +30,8 @@ import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 
-import static com.hazelcast.config.AbstractXmlConfigHelper.childElements;
-import static com.hazelcast.config.AbstractXmlConfigHelper.cleanNodeName;
+import static com.hazelcast.config.DomConfigHelper.childElements;
+import static com.hazelcast.config.DomConfigHelper.cleanNodeName;
 import static java.lang.String.format;
 
 public final class CloudyUtility {

--- a/src/main/java/com/hazelcast/cluster/impl/TcpIpJoinerOverAWS.java
+++ b/src/main/java/com/hazelcast/cluster/impl/TcpIpJoinerOverAWS.java
@@ -76,12 +76,6 @@ public class TcpIpJoinerOverAWS
     }
 
     @Override
-    protected int getConnTimeoutSeconds() {
-        AwsConfig awsConfig = fromDeprecatedAwsConfig(node.getConfig().getNetworkConfig().getJoin().getAwsConfig());
-        return awsConfig.getConnectionTimeoutSeconds();
-    }
-
-    @Override
     public String getType() {
         return "aws";
     }

